dgl_ptm/dgl_ptm/config.py

```diff
@@ -61,6 +63,8 @@ class SteeringParams(BaseModel):
     local_ratio: float = 0.25
     truncation_weight: float = 1.0e-10
     step_type: str = "default"
+    data_collection_period: int = 1
+    data_collection_list: Optional[List[int]] = None
```

Comment:
```suggestion
    data_collection_intervals: Union[int, List[int]] = 1
```

Reply:
I think this would cause confusion:
* the collection period is repeated collection after every X steps, the period.
* the collection list is an additional list of specific steps at which to collect the data.
These are very different in that the list is never repeated.

There is another way in which the name "interval" would be misleading, at least to me: I'd expect an interval to mean "every step between a specific start and end step".

Reply:
I forgot to mention that the period and list can also be used side-by-side: they could both be active at the same time.

Reply:
Here “interval” means the time interval and is commonly used to refer to regular or specified steps at which an event, such as `data_collection`, happens. The text “every n steps” in issue #91 indicates the interval between each data collection event, as it was also implemented as `timestep % params['data_collection_period'] == 0` in your code. Also, the “and/or at specified steps” implies intervals at which data collection events occur at predefined steps. We can also use words like "frequency" or "steps", but it would be less clear and precise. In any case, the usage should also be documented.

The reason for using `Union` is that there is no need to handle both cases separately, i.e. `data_collection` can be done at “every n steps **or** at specified steps”. Using a single variable for both cases makes the code more concise and easier to understand.

Reply:
We seem to be disagreeing on terminology. As I see it:
* An interval is "an intervening time", as in "the time segment between step 0 and step X". Intervals are usually defined by a start, end, and increment.
* A period is the inverse of a frequency, as "the completion of a cycle, a series of events, or a single action"
* Frequency would be the incorrect term "the rate at which something occurs over a particular period of time", because then the frequency should be 1/X. This is a very common mistake.

As for the second part of your comment:
#91 states "it would be desirable to have this every n steps **and/or** at specified steps."

I've added a comment clarifying the difference between the period and list in ptm_step.

Reply:
For an example use case of both a period and a specified list, imagine studying the effect of a policy changes. The model could run to a desired state at 1550 steps (for some reason), implement a policy change that changes some of the config parameters, and then run for another 2000 steps.

Data could be collected every 500 steps (the period) as well as at the specified steps 1550, 1551 and 3550 to see the data directly before the policy change, directly after the policy change, and at the end of the run.

Reply:
Right, helpful example. Could you please test this example as well? Currently, there are only `test_data_collection_period` and `test_data_collection_list`. It would be nice to add a test like `test_data_collection_period_list` that includes setting both parameters `period` and `list` explained in this example.